### PR TITLE
feat(avt): a11y status page and a11y status table component 

### LINKED ISSUE #3848: Component Accessibility Status: Table on Accessibility tab
#### Acceptance Criteria

- Build a custom component that can be leveraged on every component's accessibility tab's page.

image

### LINKED ISSUE #3849: Component Accessibility Status: Overview page > Accessibility status tab
#### Acceptance Criteria

- Build the following page.

> [!NOTE]  
> Not dependent on https://github.com/carbon-design-system/carbon-website/issues/3851

### LINKED ISSUE #3851: Component Accessibility Status: Overview page > Accessiblity tab > All component Status
#### Acceptance Criteria

Build the `All Component accessibility status` section on the component overview page > accessibility tab

[Link to spec](https://www.figma.com/file/tNo5wiP7cBMAMOWMkJLMHq/A11y-Component-Design?type=design&node-id=1064-63800&mode=design&t=2I2DvbXKCeJaToeo-0)

image

### PR DESCRIPTION
**Pull request by alisonjoseph**
Closes #3851
Closes #3848
Closes #3849

Building off work started in https://github.com/carbon-design-system/carbon-website/pull/3735 (should be able to close that now)

### Changelog

- New Accessibility status page `/components/overview/accessibility-status`
  - Note: added "not available" to screen reader tag list 
  - Update nav link
  - Add redirect to new url
- Update structure of data inside components.json
  - Add screen reader testing data
  - Add parent component data (to link to docs pages)
  - Add all components from testing data 
  - Add overview data, to allow us to list more components that don't appear with thumbnails on overview page (would we want a 1:1 match with thumbnails at some point?)
 - New `A11yStatus` component
 - New `A11ySTatusTag` component
 - Add `A11yStatus` component to all component a11y pages

### Testing
New page
- https://carbondesignsystem-git-a11y-status-carbon-design-system.vercel.app/components/overview/accessibility-status

Component filtered on individual component a11y pages
- https://carbondesignsystem-git-a11y-status-carbon-design-system.vercel.app/components/accordion/accessibility#accessibility-testing-status
- https://carbondesignsystem-git-a11y-status-carbon-design-system.vercel.app/components/dropdown/accessibility#accessibility-testing-status

### CODE REVIEW

File: src/pages/components/overview/accessibility-status.mdx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import { Tag } from '@carbon/react';
+import A11yStatus from 'components/A11yStatus';
+
+<PageDescription>
+
+Components are one of the key building blocks of the design system. Their
+systematic reuse helps to create visual and functional consistency across
+products.
+
+</PageDescription>
```
Comment (alisonjoseph) [inline src/pages/components/overview/accessibility-status.mdx:18]: > [!WARNING]  
> Need copy for the intro paragraph

File: src/components/A11yStatus/a11y-status.module.scss [hunk 2]
```diff
@@ -0,0 +1,13 @@
+.table {
+  margin-top: $spacing-06;
+}
+
+:global(.page-table .cds--tag) {
+  margin-left: 0;
+  white-space: nowrap;
+
+  span {
+    //override to fix alignment issue with variable font
```
Comment (alisonjoseph) [inline src/components/A11yStatus/a11y-status.module.scss:10]: TODO: check if we have latest version. alignment issues should have been fixed https://github.com/IBM/plex/issues/377

Comment (tay1orjones) [reply src/components/A11yStatus/a11y-status.module.scss:10]: I can't remember how we're bringing in Plex, but the latest version binary is too large to update. We might need to pull it in via Akamai instead for the time being.

File: src/pages/components/overview/accessibility-status.mdx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+<AnchorLink>All component accessibility status</AnchorLink>
+
+</AnchorLinks>
+
+## Accessibility tag status
+
+For every latest release, Carbon runs tests on all components to meet the
+accessibility requirements. These different statuses report the work that Carbon
+has done in the back end. These tests appear only when the components are
+stable.
```
Comment (tay1orjones) [inline src/pages/components/overview/accessibility-status.mdx]: My take at this

```suggestion
Automated accessibility verification tests are ran for each change proposed to the Carbon codebase. Additional manual tests are done periodically, such as screenreader testing. These tests ensure a consistent baseline of accessibility for each stable component. 

These accessibility tests can differ in their level of depth, breadth, and fidelity. A series of tags surface these differences throughout the website. The tags represent different levels of code test fidelity across individual components and types of testing within the system.
```

File: src/pages/components/overview/accessibility-status.mdx [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+| Tag                                     | Meaning                                                                                                                                                                                                             |
+| --------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| <Tag type="green">Tested</Tag>          | When all tests pass and there are no violations when running the automated Equal Access Checker on default state, advanced states, and keyboard navigation.                                                         |
```
Comment (tay1orjones) [inline src/pages/components/overview/accessibility-status.mdx]:
```suggestion
| <Tag type="green">Tested</Tag>          | Tests have been implemented, pass, and show no violations via the [IBM Equal Access Accessibility Checker](https://github.com/IBMa/equal-access) for a components' default state, advanced states, and keyboard navigation.                                                         |
```

File: src/pages/components/overview/accessibility-status.mdx [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+| <Tag type="blue">Partially tested</Tag> | When some tests pass or are removed, they are either combined into existing tests or substituted with new ones and moved to a different file. Skipped tests stay in the file but are clearly marked with test.skip. |
```
Comment (tay1orjones) [inline src/pages/components/overview/accessibility-status.mdx]:
```suggestion
| <Tag type="blue">Partially tested</Tag> | Some testing has been done, but is still incomplete and in progress. For example, manual tests may not have covered all use cases, or automated tests may not cover all component states. Previously existing tests may have become invalid and are temporarily skipped or removed. |
```

File: src/pages/components/overview/accessibility-status.mdx [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+| <Tag type="teal">Manually tested</Tag>  | When screen reader tests are conducted manually and validated by a human to ensure quality.                                                                                                                         |
```
Comment (tay1orjones) [inline src/pages/components/overview/accessibility-status.mdx]: Manual testing has been performed and validated by a human to ensure proper accessibility, such as screen reader testing.

Comment (tay1orjones) [inline src/pages/components/overview/accessibility-status.mdx]:
```suggestion
| <Tag type="teal">Manually tested</Tag>  | Manual testing has been performed and validated by a human to ensure proper accessibility, such as screen reader testing.                                                                                                                         |
```

File: src/components/A11yStatus/a11y-status.module.scss (hunk 2 above)
Comment (alisonjoseph) [reply src/components/A11yStatus/a11y-status.module.scss:10]: Looks like they are being pulled directly from the theme.
https://github.com/carbon-design-system/gatsby-theme-carbon/tree/main/fonts
https://github.com/carbon-design-system/gatsby-theme-carbon/blob/main/packages/gatsby-theme-carbon/src/styles/internal/_plex.scss

We could probably update that to pull from Akamai instead? I'll take a look next week.

File: src/pages/components/overview/accessibility-status.mdx [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+| <Tag>Not available</Tag>                | Usually, with keyboard navigation tests or screen reader, when this type of test is not available for certain components.                                                                                           |
```
Comment (tay1orjones) [inline src/pages/components/overview/accessibility-status.mdx]:
```suggestion
| <Tag>Not available</Tag>                | In some cases, testing data is not available for certain components or component states. For instance, keyboard navigation testing is not necessary for components that are non-interactive.                                                                                           |
```

File: src/pages/components/overview/accessibility-status.mdx [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+| <Tag type="purple">Not yet tested</Tag> | Usually, with experimental or unstable components, tests have not been conducted for either automated or manual testing.                                                                                            |
```
Comment (tay1orjones) [inline src/pages/components/overview/accessibility-status.mdx]:
```suggestion
| <Tag type="purple">Not yet tested</Tag> | Manual and/or automated testing has been deferred. Most often this applies to experimental or unstable components that do not have testing performed until they are [moved to stable](https://github.com/carbon-design-system/carbon/blob/main/docs/experimental-code.md#moving-to-stable).                                                                                            |
```

File: src/pages/components/overview/accessibility-status.mdx [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+
+## Accessibility tests
```
Comment (tay1orjones) [inline src/pages/components/overview/accessibility-status.mdx:48]:
```suggestion
## Accessibility test categories
```

File: src/pages/components/overview/accessibility-status.mdx [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+Components are one of the key building blocks of the design system. Their
+systematic reuse helps to create visual and functional consistency across
+products.
+
+</PageDescription>
+
+<AnchorLinks>
+
+<AnchorLink>Accessibility tag status</AnchorLink>
+<AnchorLink>Accessibility tests</AnchorLink>
```
Comment (alisonjoseph) [inline src/pages/components/overview/accessibility-status.mdx:23]:
```suggestion
<AnchorLink>Accessibility test categories</AnchorLink>
```

File: src/pages/components/overview/accessibility-status.mdx [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+| <Tag type="teal">Manually tested</Tag>  | When screen reader tests are conducted manually and validated by a human to ensure quality.                                                                                                                         |
+| <Tag>Not available</Tag>                | Usually, with keyboard navigation tests or screen reader, when this type of test is not available for certain components.                                                                                           |
+| <Tag type="purple">Not yet tested</Tag> | Usually, with experimental or unstable components, tests have not been conducted for either automated or manual testing.                                                                                            |
+
+## Accessibility tests
+
+Accessibility AVT (Accessibility Automated Verification Testing) involves the
+use of automated tools and software to assess the accessibility of a digital
+product against established accessibility standards, such as the Web Content
+Accessibility Guidelines (WCAG).
```
Comment (tay1orjones) [inline src/pages/components/overview/accessibility-status.mdx:53]: Every accessibility test falls into one of four categories. Marked with the tags above, these categories show the status for each of these discrete areas of testing. This additional detail helps clarify exactly what ways a component has been tested for accessibility.

Comment (tay1orjones) [inline src/pages/components/overview/accessibility-status.mdx:53]:
```suggestion
Every accessibility test falls into one of four categories. Marked with the tags above, these categories show the status for each of these discrete areas of testing. This additional detail helps clarify exactly what ways a component has been tested for accessibility.
```

File: src/pages/components/overview/accessibility-status.mdx [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+| Advanced states     | Other states rather than the default state of a component. These could be interactive states of a component or its multiple variants. | <Tag type="green">Tested</Tag><br/><Tag type="purple">Not yet tested</Tag><br/><Tag type="blue">Partially tested</Tag>                                      |
+| Keyboard navigation | This test ensures every action a mouse user can do should have a keyboard equivalent.                                                 | <Tag type="green">Tested</Tag><br/><Tag>Not available</Tag><br/><Tag type="blue">Partially tested</Tag>                                                     |
+| Screen reader       | This test ensures that the visual information on the screen is reinforced programmatically.                                           | <Tag>Not available</Tag><br/><Tag type="purple">Not yet tested</Tag><br/><Tag type="teal">Manually tested</Tag><br/><Tag type="blue">Partially tested</Tag> |
+
+## All component accessibility status
+
+This table reflects the latest accessibility testing of only the stable
+components in the React framework at the moment.
```
Comment (tay1orjones) [inline src/pages/components/overview/accessibility-status.mdx:65]:
```suggestion
This table reflects the current AVT status of stable components within `@carbon/react`.
```

File: src/pages/components/overview/accessibility-status.mdx [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+use of automated tools and software to assess the accessibility of a digital
+product against established accessibility standards, such as the Web Content
+Accessibility Guidelines (WCAG).
+
+| Test                | Meaning                                                                                                                               | Possible Tags                                                                                                                                               |
+| ------------------- | ------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Default state       | The default state of the component                                                                                                    | <Tag type="green">Tested</Tag><br/><Tag type="purple">Not yet tested</Tag><br/><Tag type="blue">Partially tested</Tag>                                      |
```
Comment (tay1orjones) [inline src/pages/components/overview/accessibility-status.mdx:57]:
```suggestion
| Default state       | Test(s) that ensure the initial render state of a component is accessible                                                                                                    | <Tag type="green">Tested</Tag><br/><Tag type="purple">Not yet tested</Tag><br/><Tag type="blue">Partially tested</Tag>                                      |
```

File: src/pages/components/overview/accessibility-status.mdx [hunk 14]
```diff
@@ -0,0 +1,62 @@
...
+| Advanced states     | Other states rather than the default state of a component. These could be interactive states of a component or its multiple variants. | <Tag type="green">Tested</Tag><br/><Tag type="purple">Not yet tested</Tag><br/><Tag type="blue">Partially tested</Tag>                                      |
```
Comment (tay1orjones) [inline src/pages/components/overview/accessibility-status.mdx:58]:
```suggestion
| Advanced states     | Tests that ensure additional states of the component are accessible. This could be interactive states of a component or its multiple variants. | <Tag type="green">Tested</Tag><br/><Tag type="purple">Not yet tested</Tag><br/><Tag type="blue">Partially tested</Tag>                                      |
```

File: src/pages/components/overview/accessibility-status.mdx [hunk 15]
```diff
@@ -0,0 +1,62 @@
...
+| Keyboard navigation | This test ensures every action a mouse user can do should have a keyboard equivalent.                                                 | <Tag type="green">Tested</Tag><br/><Tag>Not available</Tag><br/><Tag type="blue">Partially tested</Tag>                                                     |
```
Comment (tay1orjones) [inline src/pages/components/overview/accessibility-status.mdx:59]:
```suggestion
| Keyboard navigation | Tests that ensure focus is properly managed, and all interactive functions of a component have a proper keyboard-accessible equivalent.                                                 | <Tag type="green">Tested</Tag><br/><Tag>Not available</Tag><br/><Tag type="blue">Partially tested</Tag>                                                     |
```

File: src/pages/components/overview/accessibility-status.mdx [hunk 16]
```diff
@@ -0,0 +1,62 @@
...
+| Screen reader       | This test ensures that the visual information on the screen is reinforced programmatically.                                           | <Tag>Not available</Tag><br/><Tag type="purple">Not yet tested</Tag><br/><Tag type="teal">Manually tested</Tag><br/><Tag type="blue">Partially tested</Tag> |
```
Comment (tay1orjones) [inline src/pages/components/overview/accessibility-status.mdx:60]:
```suggestion
| Screen reader       | This manual testing ensures that the visual information on the screen is properly conveyed and read correctly by screen readers such as JAWS, VoiceOver, and NVDA.                                           | <Tag>Not available</Tag><br/><Tag type="purple">Not yet tested</Tag><br/><Tag type="teal">Manually tested</Tag><br/><Tag type="blue">Partially tested</Tag> |
```

**Review by tay1orjones — COMMENTED** [review]
Sorry for hijacking this a bit with my suggestions 😅  

I figured it might be best to just slot in here for the content work I needed to do in https://github.com/carbon-design-system/carbon/issues/15326

@thyhmdo could you take a look here and flag anything you think doesn't sound right?

File: src/pages/components/overview/accessibility-status.mdx (hunk 1 above)
Comment (tay1orjones) [reply src/pages/components/overview/accessibility-status.mdx:18]: My take on this:

```suggestion
<PageDescription>

Accessibility verification testing (AVT) involves the use of both manual and automated tools and software to assess the accessibility of Carbon components against established accessibility standards, such as the Web Content Accessibility Guidelines (WCAG).

Carbon implements a robust matrix of AVT to ensure that every Carbon component meets the utmost standards of accessibility. The status of this testing is outlined across the website through a detailed matrix of test categories and tags.

</PageDescription>
```

Comment (alisonjoseph) [reply src/pages/components/overview/accessibility-status.mdx:18]: This feels a bit long, when most pages are ~2 sentences for their intro text.
Screenshot 2024-01-05 at 2 14 10 PM

What about this? Maybe still too wordy?

```suggestion
<PageDescription>

Accessibility verification testing (AVT) involves the use of both manual and
automated tools and software to assess the accessibility of Carbon components
against established accessibility standards, such as the Web Content
Accessibility Guidelines (WCAG). The status of this testing is outlined across
the website through a detailed matrix of test categories and tags.

</PageDescription>
```

File: src/data/components.json [hunk 17]
```diff
@@ -1,124 +1,513 @@
...
+        "screenreader": "manual"
+      }
     },
     {
-      "component": "Slider"
+      "component": "Fluid textinput",
+      "parentComponent": "Textinput",
+      "overview": false,
+      "testing": {
+        "screenreader": "manual"
```
Comment (alisonjoseph) [inline src/data/components.json:212]: @andreancardona have our fluid components been screen reader tested?

Comment (andreancardona) [reply src/data/components.json:212]: unstable ones no. stable fluid yes!

Comment (alisonjoseph) [reply src/data/components.json:212]: Thanks, updated the data to reflect that. I believe only fluid textinput is stable from what I can see.

**Comment by mbgower** [comment]
This is a design question, not implementation question, so feel free to get me to repost elsewhere, but reviewing this, I had the following question:
Do we really need "Manually tested?

First, it is the longest tag, so it's much easier to design without it. Second, it implies that no keyboard testing is manually done, which I don't think is accurate. Third, I don't see what we lose by just having the 'pass' result of screen reader be "tested".
image

I suspect this has already been raised, and there is a good response, but I was struck by it during my review of the work. Thanks!
BTW, if this suggestion was adopted, there would have to be some slight wording in the meaning column:
> Tests have been implemented, pass, and show no violations, either via the [IBM Equal Access Accessibility Checker](https://github.com/IBMa/equal-access) for a components’ default state, advanced states, or via manual testing.

image

**Comment by thyhmdo** [comment]
> This is a design question, not implementation question, so feel free to get me to repost elsewhere, but reviewing this, I had the following question: Do we really need "Manually tested?
> 
> First, it is the longest tag, so it's much easier to design without it. Second, it implies that no keyboard testing is manually done, which I don't think is accurate. Third, I don't see what we lose by just having the 'pass' result of screen reader be "tested". image
> 
> I suspect this has already been raised, and there is a good response, but I was struck by it during my review of the work. Thanks! BTW, if this suggestion was adopted, there would have to be some slight wording in the meaning column:
> 
> > Tests have been implemented, pass, and show no violations, either via the [IBM Equal Access Accessibility Checker](https://github.com/IBMa/equal-access) for a components’ default state, advanced states, or via manual testing.
> 
> BTW, on reflection, since you break each component down by its default and advanced states, keyboard, and screen reader, there's no need to state these so thoroughly in the meaning, IMO. I also think a bit of the wording at the front is unnecessary, so I suggest:
> 
> > **Tests have been carried out, via either the [IBM Equal Access Accessibility Checker](https://github.com/IBMa/equal-access) or manual testing, and show no violations.**
> 
> image

In general, we want to make it clear for non-technical users or those who are not familiar with this to understand that other tests are automatic while Screen reader requires a manual process so that's why it has Manually tested tag. 

And yes, we still need some help on that wording. That's a good suggestion! Thanks @mbgower

**Comment by mbgower** [comment]
> other tests are automatic while Screen reader requires a manual process

Keyboard is also often (but not always?) going to be manual too, though, right? I think the existing generic wording of "Partially tested" supports making "Tested" also generic.

> we want to make it clear for non-technical users or those who are not familiar with this to understand

I think you can provide the kind of context you're talking about in the test categories table. I would argue that the "Manually tested" wording makes the results busier without really being clearer ; I'd argue it actually makes it a bit more confusing.
image

**Comment by thyhmdo** [comment]
Looks great! Here are a few things I have: 

1. The underline has a problem in the smaller breakpoint, under Keyboard navigation. 

image



2. For the text and the underline in the tag, can we somehow have them closer together with the text aligned more to the center? 

image

3. I'd like to add a tooltip next to the "Accessibility testing status" on both Individual component page and the Overview page. I think we can improve the text but that's the idea. 

image

4. I'm using Arc, and I know a lot more people are using this browser nowadays, the tag wrapping is still posing the issue. Is this something we can support as well? 

image

**Comment by aubrey-oneal** [comment]
Hey @thyhmdo and @tay1orjones I spent some time drafting possible tooltips descriptions for your tags. Based on your designs, it seems this tooltip is always in the context of a component's state, so I did utilize that context. I also tried to clarify that 'Tested' implies automated testing—something that took me a little while to understand. I've _really_ paired down the info, probably more than I should. Please add back in what you need. My ideal character count for the tooltips would be 70 characters, 90 max. For Taylor's original 'meaning definitions' I only changed a word here or there.

**Tested**
- Meaning: Automated tests have been implemented, pass, and show no violations via the IBM Equal Access Accessibility Checker for a component’s default state, advanced states, and keyboard navigation.
- Tooltip (still too long): Passes automated tests and shows no violations via the IBM Equal Access Accessibility Checker. 

**Partially tested**
- Meaning: Some testing has been done, but is still incomplete and in progress. For example, manual tests may not have covered all use cases, or automated tests may not cover all component states. Previously existing tests may have become invalid and are temporarily skipped or removed.
- Tooltip: Some tests are incomplete, in progress, invalid, or temporarily skipped.

**Manually tested**
- Meaning: Manual testing has been performed and validated by a human to ensure proper accessibility, such as screen reader testing.
- Tooltip: A human has manually tested this component, e.g. screen reader testing.

**Not available**
- Meaning: In some cases, testing data is not available for certain components or component states. For instance, keyboard navigation testing is not necessary for components that are non-interactive.
- Tooltip: Test data is either not available or not applicable for this component state.

**Not yet tested**
- Meaning: Automated or manual testing has been deferred. Most often this applies to experimental or unstable components that do not have testing performed until they are moved to stable.
- Tooltip (still too long): Automated or manual testing has been deferred until this component state is stable.

**Comment by tay1orjones** [comment]
> Keyboard is also often (but not always?) going to be manual too, though, right?

Once a component has been marked stable and had automated keyboard navigation tests written, it's actually the inverse because that testing is done on every pull request.

The tests simulate pressing tab and ensure the intended elements get focus in the proper order as documented on the a11y pages on the website for each component. Manual tests are done as well but it's less frequent/inconsistent.

**Comment by tay1orjones** [comment]
> Keyboard is also often (but not always?) going to be manual too, though, right?

@mbgower Once a component has been marked stable and had automated keyboard navigation tests written, it's actually the inverse because that automated testing is done on every pull request.

The tests simulate pressing tab and ensure the intended elements get focus in the proper order as documented on the a11y pages on the website for each component. Manual tests are done as well but it's less frequent/inconsistent.

**Comment by tay1orjones** [comment]
@aubrey-oneal Thank you! These look really great. Here's my take:

#### Tested
```diff
- Tooltip (still too long): Passes automated tests and shows no violations via the IBM Equal Access Accessibility Checker.
+ Tooltip: Passes all automated tests with no reported accessibility violations.
```

#### Not yet tested
```diff
- Tooltip (still too long): Automated or manual testing has been deferred until this component state is stable.
+ Tooltip: Automated or manual testing has been temporarily deferred.
```

**Comment by mbgower** [comment]
> > Keyboard is also often (but not always?) going to be manual too, though, right?
> 
> @mbgower Once a component has been marked stable and had automated keyboard navigation tests written, it's actually the inverse because that automated testing is done on every pull request.
> 
> The tests simulate pressing tab and ensure the intended elements get focus in the proper order as documented on the a11y pages on the website for each component. Manual tests are done as well but it's less frequent/inconsistent.

Okay, but there still does not seem to be a lot of value having the "Manually tested" tag, especially given even with your qualifications, there can still be manual tests done for keyboard. Yet at no time do I see anyone using "manually tested" for keyboard.

In short, I feel like the wording suggestions I am proposing would reduce possible confusion, reduce the number of tags that can be used, and create a more consistent message.

**Comment by tay1orjones** [comment]
@mbgower I think the primary reasons for having a `Manually tested` tag are:

1. Informing users and stakeholders we're taking the time to actually do manual verification of these things
2. It's a separate process with different requirements than automated testing (different screenreaders could be used (or not), quality of testing depends on the tester and their methods, not all components are manually tested at the same time, it's only partially done on every PR, etc.)
3. Manual testing is not done as often as automated testing, there is greater possibility for gaps in coverage or things to break between manual tests.
4. Manual testing does not produce the same artifacts that are available for automated tests (links to test source code, for example)


With this in mind, maybe `Tested` is flawed like you suggested earlier and should be renamed to `Automatically tested` or something.

**Comment by mbgower** [comment]
Here is the information provided in the descriptions of testing
image

There is no reason we could not elaborate more fully on some of the points you make within the description of Screen Reader testing -- which already says "manually". You'll note that the keyboard description at least infers there is some manual testing in some situations. I still don't see _what_ value there is in slapping "Manually tested" as a tag. Also, I'd argue that if by SR tests, we mean testing for the appropriate name, role, value; more of those are caught by automated tests all the time. So we could even add some nuance to the discussion there.

I'm also not sure about "Not available" and the combinations that are used of it. I like the combination that is used for keyboard, and i think the same combination could be used for Screen Reader. The following description infers it could even take place with advanced states. By the same token, I'm not sure why "Not tested" isn't showing up for Keyboard.

> In some cases, testing data is not available for certain components or component states. For instance, keyboard navigation testing is not necessary for components that are non-interactive.

**Comment by alisonjoseph** [comment]
@thyhmdo I've fixed all the wrapping issues and added the help icon. I skipped number 2, the text alignment. I have a [PR open in the gatsby theme](https://github.com/carbon-design-system/gatsby-theme-carbon/pull/1395) to update to the latest version of Plex(variable) which adjusts some of the alignment issues, and will hopefully fix that. 

Also as a FYI the link color is off in the toggletip, we have a couple open issues in core Carbon around this. 
https://github.com/carbon-design-system/carbon/issues/14732
https://github.com/carbon-design-system/carbon/issues/15204
Screenshot 2024-01-16 at 2 11 24 PM

@tay1orjones @aubrey-oneal I updated the copy for the descriptions.

File: src/pages/components/overview/accessibility-status.mdx (hunk 1 above)
Comment (aubrey-oneal) [reply src/pages/components/overview/accessibility-status.mdx:18]: The page description does need to be short and I noticed both of your options had information that  was similar to the Accessibility tag status content. So here is an option where I massaged some of your messaging underneath that header:

Page description:
Manual and automated tests assess the accessibility of Carbon components against established standards. Read on to learn about how components are tagged with test statuses across the website.

Accessibility tag status

Automated accessibility verification tests (AVT) are run for each change proposed to the Carbon codebase. Additional manual tests are done periodically, such as screenreader testing. These tests ensure a consistent baseline of accessibility for each stable component.

Carbon implements a robust matrix of AVT to ensure that every Carbon component meets the utmost standards of accessibility. The status of this testing is outlined across the website through a detailed matrix of test categories and tags, outlined below.

**Comment by thyhmdo** [comment]
Looking great!! Just a few minor things


---
This is on me, but I'm looking to change the text to:

> Learn more about tag and test meaning
> View all component accessibility status

image

---
Change to with the lower case of "tags"

image


----
I noticed that this [one variant page](https://carbondesignsystem-git-a11y-status-carbon-design-system.vercel.app/components/accordion/accessibility#accessibility-testing-status) "View all component..." link takes me to the "Accessibility tag status" table instead of all the "All component accessibility status" section.              
While the [multiple variant page](https://carbondesignsystem-git-a11y-status-carbon-design-system.vercel.app/components/dropdown/accessibility#accessibility-testing-status) takes me to half of "Accessibility test categories" table?

File: src/pages/components/overview/accessibility-status.mdx (hunk 1 above)
Comment (alisonjoseph) [reply src/pages/components/overview/accessibility-status.mdx:18]: Thanks!

**Comment by alisonjoseph** [comment]
@thyhmdo just pushed a fix. The anchor links weren't working correctly because of the tooltip within the title.

**Review by thyhmdo — APPROVED** [review]
Looks great!

**Comment by guidari** [comment]
@alisonjoseph One question.

When I try to access this component "Aspect Ratio" I get an 404. I know it is because we don't have a page in the website for the component so the error is totally right. But just make sure, should we cover that somehow or that's okay?
Might hard to filter these, idk...
[Link to access
](https://carbondesignsystem-git-a11y-status-carbon-design-system.vercel.app/components/overview/accessibility-status/#:~:text=Manually%20tested-,Aspect%20ratio,-Default%20state
)

**Comment by alisonjoseph** [comment]
@guidari ah good catch, I'll fix that.

**Comment by alisonjoseph** [comment]
fixed @andreancardona

**Review by andreancardona — APPROVED** [review]
🔥

**Comment by mbgower** [comment]
Obviously kinda late for this comment, but I was a little surprised to see these appearing at the top of the accessibility tab.
Historically they've been near the bottom.  i guess I was focused on terminology and hadn't seen the intended location on the Accessibility tab in prior work -- or maybe I thought it was going to positioned in the right margin on the main page or something?

I did want to flag that it's a little odd that these are implemented with an H3, since the page structure is now wonky:
H3 Accessibility testing status
H2 What Carbon provides
H3 Keyboard interactions
...
H2 Development considerations

I don't think the status deserves an H2 -- its not on par with the other typical H2s:

- What Carbon provides
- Design recommendations
- Development considerations

IMO, it conceptually probably makes the most sense under Development considerations.

**Comment by thyhmdo** [comment]
> @thyhmdo Obviously kinda late for this comment, but I was a little surprised to see these appearing at the top of the accessibility tab. Historically they've been near the bottom. i guess I was focused on terminology and hadn't seen the intended location on the Accessibility tab in prior work -- or maybe I thought it was going to positioned in the right margin on the main page or something?
> 
> I did want to flag that it's a little odd that these are implemented with an H3, since the page structure is now wonky: H3 Accessibility testing status H2 What Carbon provides H3 Keyboard interactions ... H2 Development considerations
> 
> I don't think the status deserves an H2 -- its not on par with the other typical H2s:
> 
> * What Carbon provides
> * Design recommendations
> * Development considerations
> 
> IMO, it conceptually probably makes the most sense under Development considerations.

Hey @mbgower thanks for the feedback and noted. We probably can change the order of it in the future. We also moved away from the margin space because of the condensed space, this will soon be reflected on tiles on the Usage tab work instead.
